Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java

```diff
@@ -1,298 +1,374 @@
- * Mockito.doReturn(true).when(operations).pushToKafka(any(), any(), any());
- * Mockito.doThrow(new
- * ResponseException(ErrorType.AlreadyExists)).when(entityService).createMessage
- * (any(), any()); entityService.createMessage(any(), entityPayload);
- * verify(entityTopicMap, times(1)).isExist(any());
- * 
- * }
- * 
- *//**
-	 * this method is use for update the entity
-	 * 
-	 * @throws Exception
-	 */
-/*
- * @Test public void updateMessageTest() throws Exception{
- * 
- * EntityDetails entityDetails=Mockito.mock(EntityDetails.class); byte[]
- * messageByte=entityPayload.getBytes(); JsonNode resultJson =
- * objectMapper.createObjectNode(); UpdateResult updateResult = new
- * UpdateResult(updateJsonNode, resultJson); updateResult.setStatus(true);
- * 
- * Mockito.doReturn(entityDetails).when(entityTopicMap).get(any());
- * Mockito.doReturn(messageByte).when(operations).getMessage(any(), any(),
- * any(Integer.class), any(Long.class));
- * Mockito.doReturn(payloadNode).when(objectMapper).readTree(any(String.class));
- * Mockito.doReturn(blankNode).when(objectMapper).createObjectNode(); //
- * Mockito.doReturn(updateResult).when(entityService).updateFields(messageByte,
- * updateJsonNode, null); //TODO no assert. no usage of result
- * Mockito.doReturn(updateResult).when(entityService).updateMessage(any(),
- * any(), any()); entityService.updateMessage(any(), "urn:ngsi-ld:Vehicle:A103",
- * updatePayload); verify(entityService, times(1)).updateMessage(any(),
- * any(),any());
- * 
- * 
- * }
- * 
- *//**
+	@Test
+	public void createMessageTest() {
+		try {
+			// TODO redo logic has changed no quick fix
+			/*
+			 * HttpServletRequest mockedRequest = Mockito.mock(HttpServletRequest.class);
+			 * MockHttpServletRequest mockRequest = new MockHttpServletRequest();
+			 * mockRequest.addHeader("Accept", "application/json");
+			 * mockRequest.addHeader("Content-Type", "application/json");
+			 * mockRequest.addHeader("Link",
+			 * "<https://raw.githubusercontent.com/easy-global-market/ngsild-api-data-models/feature/add-json-ld-context-for-ngsi-ld-test-suite/ngsi-ld-test-suite/ngsi-ld-test-suite-context.jsonld>; rel=\"http://www.w3.org/ns/json-ld#context\"; type=\"application/ld+json\""
+			 * );
+			 */
+			multimaparr.put("content-type", "application/json");
+			Gson gson = new Gson();
+			Map<String, Object> resolved = gson.fromJson(entityPayload, Map.class);
+			String id = entityService.createEntry(multimaparr, resolved);
+
+			Assert.assertEquals(id, "urn:ngsi-ld:Vehicle:A103");
+
+		} catch (Exception ex) {
+			Assert.fail();
+		}
+
+	}
+
+	@Test
+
+	public void createMessageThrowsAlreadyExistTest() throws ResponseException, Exception {
+		multimaparr.put("content-type", "application/json");
+		Gson gson = new Gson();
+		Map<String, Object> resolved = gson.fromJson(entityPayload, Map.class);
+		Mockito.doThrow(new ResponseException(ErrorType.AlreadyExists, " already exists")).when(entityService)
+				.createEntry(any(), any());
+
```

Comment:
[Major] This test case is not complete because need to compare actual and expected result also.